src/OpenEventSourcing.RabbitMQ/Management/Api/RabbitMqManagementApiClient.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using OpenEventSourcing.RabbitMQ.Extensions;
+using OpenEventSourcing.RabbitMQ.Management.Api.Responses;
+
+namespace OpenEventSourcing.RabbitMQ.Management.Api
+{
+    public class RabbitMqManagementApiClient : IRabbitMqManagementApiClient
+    {
+        private readonly HttpClient _client;
```

Comment:
Is it worth using the httpfactory stuff here? I know we register the httpclient using the `services.AddHttpClient` extension, but just wondering if its worth going with the http client factory pattern and just newing up a new client on demand in the `RetrieveSubscriptionsAsync` method? then you remove the need to inject `IOptions<RabbitMqOptions>` as the http client config would be set at a higher level.

Not sure if it will bring much benefit, just a suggestion.

Reply:
This is using the HttpClientFactory package. See the docs [here](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.0#typed-clients)

Reply:
Ah my bad 🙈

Reply:
Don't think so, what would this achieve?